### ISSUE TITLE
Adding XGBoost model to official benchmark models

### ISSUE THREAD

**Issue by haozhenWu** [deepchem/deepchem#462]
Hi, @rbharath 
I saw #433. I am currently trying out the new dc.molnet module. Just want to ask before I start to prepare the PR. 
Will you be interested to add XGBoost model into official models? Or does someone else have already been doing this? Based on my preliminary test, XGBoost model has comparative result compared to Logistic regression and Random Forest. 

Thanks.

**Comment by rbharath** [comment]
@haozhenWu Thanks for raising the issue! No one else is currently working on XGBoost additions. I'd be open to a PR, but only if we can get it running on Travis-CI easily. Perhaps if there's an XGBoost anaconda package we can use.

**Comment by haozhenWu** [comment]
Thanks @rbharath . Working on the PR as soon as possible.

**Comment by mrwns** [comment]
I was planning to do that after handing in my PhD thesis ( in a couple of weeks ), as there is some evidence [(Sheridan) http://pubs.acs.org/doi/abs/10.1021/acs.jcim.6b00591, and https://arxiv.org/pdf/1701.01329.pdf ] that xgboost can be better than Random Forest and single task neural nets. GraphConvs however will most likely be still better, let's find out. If you have some spare time to do the benchmarks now please go ahead @haozhenWu , then I don't have to :)

**Comment by haozhenWu** [comment]
@mrwns Thanks for your evidence. I actually have been working on PCBA, DUDE and Tox21 datasets since last year and built up my own package [lightchem](https://github.com/haozhenWu/lightchem), which features a stacking approach for ligand-based virtual screening. I have a chance to [benchmark](https://github.com/haozhenWu/lightchem#benchmark-results) these datasets using single XGBoost model (both tree booster and linear boost). However I used stratified split so that the results are not directly comparable. 
@rbharath This might be a separate topic. In the future, will you also be interested in the stacking approach? For example, for PCBA datasets, I think DeepChem so far only provides binary label. But the raw datasets actually come with continuous score( like IC50 score, PUBCHEM activity score). Lightchem basically use both label types and multiple featurization methods to train a stacking model. Based on stratified split benchmark of PCBA128, lightchem model does improve performance compared to best single model.

**Comment by rbharath** [comment]
The latest version of MoleculeNet has XGBoost benchmarks so going to close this issue.
